tests/test_jointcal_hsc.py

```diff
@@ -95,27 +95,20 @@ def test_jointcalTask_11_visits_no_photometry(self):
         dist_rms_relative = 17e-3*u.arcsecond
         pa1 = None  # pa1 = 0.134
         metrics = {'collected_astrometry_refStars': 3649,
-                   'collected_photometry_refStars': 3649,
-                   'selected_astrometry_refStars': 3649,
-                   'selected_photometry_refStars': 3649,
+                   'selected_astrometry_refStars': 1038,
                    'associated_astrometry_fittedStars': 2908,
-                   'associated_photometry_fittedStars': 2908,
                    'selected_astrometry_fittedStars': 2203,
-                   'selected_photometry_fittedStars': 2203,
                    'selected_astrometry_ccdImages': 33,
-                   'selected_photometry_ccdImages': 33,
                    'astrometry_final_chi2': 7929.656,
                    'astrometry_final_ndof': 14262,
-                   'photometry_final_chi2': None,
-                   'photometry_final_ndof': None
```

Comment:
Why the drastic pruning? It also makes me realize that some tests have all these metrics and others do not. Some explanation as a code comment would be welcome when standard metrics are missing. If it's a matter of needing a looser tolerance, this suggests the test framework is too rigid and should support a number and a tolerance, though that is arguably out of scope for this ticket.

Reply:
This particular change was removing metrics that weren't being measured. Some tests are photometry/astrometry only, so they don't have the other metrics.

`JointcalTestBase._test_metrics()` loops over the measurements, so if something is being measured but doesn't have an expected value, the test will fail, unless explicitly defined as `None`.